arch/arc/core/fault_s.S

```diff
@@ -205,21 +205,23 @@ SECTION_SUBSEC_FUNC(TEXT,__fault,__ev_trap)
 	mov_s r6, K_SYSCALL_BAD
 
 valid_syscall_id:
+	_create_irq_stack_frame
+
```

Comment:
I would add here a comment explaining that we save `r0` in `_create_irq_stack_frame()` and so now we may safely use it for other things with no fear or losing its initial contents.

Reply:
done

Reply:
Well it's obvious to you and me that caller regs are saved in IRQ stack frame but new reader might be confused.
That said maybe explicitly mention that `r0` is saved in `_create_irq_stack_frame()`?

Also looks like comments indentation in other places of the same source file is different (+1 tab) compared to ones above and below this place.